libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/LegacyEditorFragment.java

```diff
@@ -966,53 +966,58 @@ public void onResponse(ImageLoader.ImageContainer container, boolean arg1) {
     }
 
     @Override
-    public void appendMediaFile(MediaFile mediaFile, String imageUrl, ImageLoader imageLoader) {
-        WPEditImageSpan imageSpan = createWPEditImageSpan(getActivity(), mediaFile);
+    public void appendMediaFile(final MediaFile mediaFile, final String imageUrl, final ImageLoader imageLoader) {
+        mActivity.runOnUiThread(new Runnable() {
```

Comment:
A lot of the work is still being done on the main thread. It does seem to load faster and profiling showed a slight improvement of ~7% (decrease in time spent on the main thread) in the one comparison I ran. There is still a noticeable delay if an absurd amount of images are selected but very few people will run into that problem.